iohk/UpdateProposal.hs

```diff
@@ -270,33 +282,47 @@ instance Checkable UpdateProposalConfig1 where
     | otherwise = Nothing
 
 instance Checkable UpdateProposalConfig2 where
-  checkConfig (UpdateProposalConfig2 p r) = checkConfig p <|> check r
+  checkConfig (UpdateProposalConfig2 p r b s) = checkConfig p <|> check r
+                                                <|> checkInstallerHashes b
+                                                <|> checkInstallerHashes s
     where
       check InstallersResults{..}
         | grApplicationVersion <= 0 = Just "Application version must be set"
         | T.null grCardanoCommit = Just "Missing cardano commit"
         | missingVersion Mac64 ciResults = Just "macOS version is missing"
         | missingVersion Win64 ciResults = Just "Windows version is missing"
+        | missingVersion Linux64 ciResults = Just "Linux version is missing"
+        | length (nub as) /= length as = Just "Multiple installers found for an arch"
         | otherwise = Nothing
         where
           GlobalResults{..} = globalResult
           missingVersion arch = not . any ((== arch) . ciResultArch)
+          as = map ciResultArch ciResults
```

Comment:
I think the above code could've been simplified if `ciResults` was an `ArchMap`, making the structural checks unnecessary.